src/BeamAdapter/component/model/WireSectionMaterial.h

```diff
@@ -0,0 +1,78 @@
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: see Authors.md                                                     *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <BeamAdapter/config.h>
+#include <BeamAdapter/utils/BeamSection.h>
+#include <sofa/core/objectmodel/BaseObject.h>
+
+namespace sofa::beamadapter
+{
+
+/**
+ * \class WireRestShape
+ * \brief Describe the shape functions on multiple segments
+ *  
+ *  Describe the full shape of a Wire with a given length and radius. The wire is discretized by a set of beams (given by the keyPoints and the relatives Beam density)
+ *  This component compute the beam discretization and the shape functions on multiple segments using curvilinear abscissa.
+ */
+class SOFA_BEAMADAPTER_API WireSectionMaterial : public core::objectmodel::BaseObject
+{
+public:
+    SOFA_CLASS(WireSectionMaterial, core::objectmodel::BaseObject);
+
+    /// Default Constructor
+    WireSectionMaterial();
+
+
+    void init();
+
+    /// This function gives the Young modulus and Poisson's coefficient of the beam depending on the beam position
+    void getYoungModulusAtX(float& youngModulus, float& cPoisson);
+    void getYoungModulusAtX(double& youngModulus, double& cPoisson);
+
+    /// This function gives the mass density and the BeamSection data depending on the beam position
+    void getInterpolationParam(float& _rho, float& _A, float& _Iy, float& _Iz, float& _Asy, float& _Asz, float& _J);
+    void getInterpolationParam(double& _rho, double& _A, double& _Iy, double& _Iz, double& _Asy, double& _Asz, double& _J);
+
+    [[nodiscard]] int getNbVisualEdges() const { return d_nbEdgesVisu.getValue(); }
+
+    [[nodiscard]] int getNbCollisionEdges() const { return d_nbEdgesCollis.getValue(); }
+     
+
+   
+    /// User Data about the Young modulus
+    Data<SReal> d_poissonRatio;
+    Data<SReal> d_youngModulus;
+
+    /// Radius
+    Data<SReal> d_radius;
+    Data<SReal> d_innerRadius;
+    Data<SReal> d_massDensity;
+
+    Data< int > d_nbEdgesVisu;
+    Data< int > d_nbEdgesCollis;
```

Comment:
This PR will be very breaking so lets change for better type if possible 🥸
so unsigned type for size would be better (I dont think there are cases like using "negative" values but I may be wrong)